File: src/main/java/se/lars/types/Quotes.java
package se.lars.types;


import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import org.immutables.value.Value;

import java.util.*;

@Value.Immutable
@JsonSerialize(as = ImmutableQuotes.class)
@JsonDeserialize(as = ImmutableQuotes.class)
public interface Quotes
{
    Optional<Date> lastUpdated();
    OptionalDouble openPrice();
    OptionalDouble lastPrice();
    OptionalDouble askPrice();
    OptionalDouble bidPrice();
    OptionalDouble highPrice();
    OptionalDouble lowPrice();
    OptionalDouble tradedVolume();
    OptionalDouble tradedAmount();
}

File: src/main/kotlin/se/lars/types/Quotes.kt
package se.lars.types



import java.util.*

data class Quotes(val lastUpdated: Date?,
                  val openPrice: Double?,
                  val lastPrice: Double?,
                  val askPrice: Double?,
                  val bidPrice: Double?,
                  val highPrice: Double?,
                  val lowPrice: Double?,
                  val tradedVolume: Double?,
                  val tradedAmount: Double? )


